src/Sentry/Internal/DebugStackTrace.cs

```diff
@@ -88,25 +91,55 @@ internal void MergeDebugImagesInto(SentryEvent @event)
         Dictionary<string, string> relocations = new();
         for (var i = 0; i < DebugImages.Count; i++)
         {
-            // First check if the image is already present on the event. Simple lookup should be faster than
-            // constructing a map first, assuming there are normally just a few debug images affected.
-            var found = false;
-            for (var j = 0; j < originalCount; j++)
+            // Managed debug images
+            if (DebugImages[i].ModuleVersionId is not null)
             {
-                if (DebugImages[i].ModuleVersionId == @event.DebugImages[j].ModuleVersionId)
+                // First check if the image is already present on the event. Simple lookup should be faster than
+                // constructing a map first, assuming there are normally just a few debug images affected.
+                var found = false;
+                for (var j = 0; j < originalCount; j++)
                 {
-                    if (i != j)
+                    if (DebugImages[i].ModuleVersionId == @event.DebugImages[j].ModuleVersionId)
                     {
-                        relocations.Add(GetRelativeAddressMode(i), GetRelativeAddressMode(j));
+                        if (i != j)
+                        {
+                            relocations.Add(GetRelativeAddressMode(i), GetRelativeAddressMode(j));
+                        }
+                        found = true;
+                        break;
                     }
-                    found = true;
+                }
+
+                if (!found)
+                {
+                    relocations.Add(GetRelativeAddressMode(i), GetRelativeAddressMode(@event.DebugImages.Count));
```

Comment:
Why `@event.DebugImages.Count` here?

Reply:
because that's the index we're adding the image at on the next line.